### ISSUE TITLE
A crash occurs when attempting to access the Flipper UI endpoint

### ISSUE THREAD

**Issue by gakalantzis** [jnunemaker/flipper#607]
Hey, thanks so much for all your hard work on this awesome gem!  My team and I have run into a really bizarre sticking point with Flipper that happens very occasionally in our production environment, but consistently in our staging environment.  When we go to access the Flipper UI endpoint, we receive the default Rails error message.

We've added a class in the flipper initializer so as to lock down the endpoint to admin users only:

`class CanAccessFlipperUI
  def self.matches?(request)
    current_user = request.env['warden'].user
    current_user.present? && current_user.respond_to?(:admin?) && current_user.admin?
  end
end`

And we mount the route like so:

`constraints CanAccessFlipperUI do
    mount Flipper::UI.app(Flipper) => 'flipper'
  end`
  
  The Staging application runs on Heroku, however, the log doesn't give a whole lot to go on:
  
  `2022-02-09T21:00:08.636235+00:00 heroku router: at=info method=GET path="/flipper/features?gdpr=0" host=***.***.com request_id=0e42e498-7ae6-4e76-8f1b-da32e85ca1c6 fwd="***.**.**.*" dyno=web.1 connect=0ms service=13ms status=404 bytes=1795 protocol=https`
  
We're also running a Staging instance on AWS and are running into the exact same issue on there.  Unfortunately, those logs don't show any errors whatsoever.  It just crashes.

We use Devise to handle all of our sessions.  We're also running the app on a Unicorn server.  We're using Ruby 2.6.7 and Rails 6.0.4.1.  This issue doesn't occur when I run the app locally.  I searched through all the previous issues that users had when trying to access the Flipper UI, however none of the suggestions there helped at all.  I've also updated Flipper to utilize the latest version, 0.23.1.  Any help would be greatly appreciated, thanks!

**Comment by jnunemaker** [comment]
@gakalantzis you are welcome. I'm struggling to think of anything that would cause this. The UI just gets mounted like any other thing in your routes file. 

Does the same happen when you mount blazer or pghero or any other engines? 
Does only flipper have a problem? Everything else in your app works? 
Do you have any error tracking in the app other than what gets put in logs? 

I think I'd almost have to hop on a call with you where we could screenshare to see more.

**Comment by jnunemaker** [comment]
Closing to keep things tidy but if we can figure out this is a flipper bug then I'll re-open it.

**Comment by silent-e** [comment]
I am also getting a 500 error when trying to use the UI in my staging environment.  I created a quick spike branch to test blazer and that engine's UI is running fine in staging.  I've removed any constraints from my flipper UI url.  I'm using the ActiveRecord adapter on this app.  I have a second app where the UI works fine in staging and that's using the Redis adapter.

When tailing my staging log and trying to load the features page it ends abruptly at

> D, [2022-08-31T12:18:20.994102 #20473] DEBUG -- : [abb21af0-5ce4-4c35-a88b-17b82174b48b] [uid: 155]    (0.6ms)  SELECT `flipper_features`.`key` AS feature_key, `flipper_gates`.`key`, `flipper_gates`.`value` FROM `flipper_features` LEFT OUTER JOIN `flipper_gates` ON `flipper_features`.`key` = `flipper_gates`.`feature_key`

There's no `Rendered …` or `Completed …` lines with allocation information, no stacktrace, no error output.  It just dies.

I'm continuing my research.

ENV:
Rails 7.0.3.1
Ruby 3.0.4
Flipper 0.25.2
Flipper ActiveRecord 0.25.2

(e)
